src/librustc_codegen_ironox/type_.rs

```diff
@@ -46,25 +47,46 @@ pub enum LLType {
         args: Vec<Type>,
         ret: Type
     },
-    /// A scalar type.
+    Array {
+        length: u64,
+    },
+    PtrTo {
+        pointee: Type,
+    },
     Scalar(ScalarType),
-    /// A placeholder for types. This will be removed later.
-    None,
+    StructType {
+        name: Option<String>,
+        members: Vec<Type>
+    },
+    Void,
 }
 
 impl CodegenCx<'ll, 'tcx> {
-    crate fn type_named_struct(&self, name: &str) -> &Type {
-        unimplemented!("type_named_struct");
+    crate fn type_named_struct(&self, name: &str) -> Type {
+        let mut borrowed_types = self.types.borrow_mut();
```

Comment:
Is `borrowed_*` something you saw in the LLVM backend? If not, I'd kill the `borrowed_` prefix.

Reply:
Done.